Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/squash.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (C) 2022 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import logging
+import re
+import sys
+import os
+import re
+from unittest import result
+
+from .command import Command
+from .branch import Branch
+from .pull_request import PullRequest
+
+from webkitbugspy import Tracker
+from webkitcorepy import arguments, run, Terminal
+from webkitscmpy import local, log, remote
+from ..commit import Commit
+
+
+class Squash(Command):
+    name = 'squash'
+    help = 'Squash from current '
+    REVERT_TITLE_TEMPLATE = 'Revert [{}] {}'
+    REVERT_TITLE_RE = re.compile(r'^Revert \[{}\] [\s\S]*'.format(Commit.IDENTIFIER_RE.pattern))
+
+    @classmethod
+    def parser(cls, parser, loggers=None):
+        PullRequest.parser(parser, loggers=loggers)
+        group = parser.add_mutually_exclusive_group(required=False)
+        group.add_argument(
+            '--interactive',
+            dest='interactive',
+            default=False,
+            action='store_true',
+            help='Use git rebase to interactivly to select what commits you want to squash.'
+        )
+
+        group.add_argument(
+            '--no-sub-commit-message',
```

Comment:
Feels like we need a better and shorter argument here...maybe `--new-message`? Could even keep the more verbose one as an alternate.